src/test/java/oof/command/ScheduleCommandTest.java

```diff
@@ -0,0 +1,106 @@
+package oof.command;
+
+import oof.Oof;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import oof.exception.ParserException;
+import oof.exception.command.CommandException;
+import oof.model.task.Event;
+import oof.model.task.Deadline;
+import oof.model.task.Task;
+import oof.model.task.TaskList;
+import oof.model.task.Todo;
+import org.junit.jupiter.api.Test;
+
+//@@author debbiextan
+
+public class ScheduleCommandTest {
+
+    private Oof oof = new Oof();
+    private TaskList taskList = oof.getTaskList();
+
+    /**
+     * Tests behavior when no date is given.
+     */
+    @Test
+    public void execute_dateEmpty_exceptionThrown() {
+        try {
+            oof.executeCommand("schedule");
+            fail();
+        } catch (CommandException | ParserException e) {
+            assertEquals("OOPS! Please enter a date!", e.getMessage());
+        }
+    }
+
+    /**
+     * Tests behavior of displaying schedule by date when there are no tasks for that date.
+     *
+     * @throws CommandException if command is invalid.
+     * @throws ParserException  if command cannot be parsed.
+     */
+    @Test
+    public void execute_correctDescriptionWithoutTasks_scheduleShown() throws CommandException, ParserException {
+        String date = "01-01-2020";
+        oof.executeCommand("schedule " + date);
+        TaskList scheduleTasks = new TaskList();
```

Comment:
Do not misuse TaskList. Use an arraylist of Tasks if you want to track the number of tasks.